### PR TITLE
feat: implement create external table parser

### PR DESCRIPTION
**Pull request by WenyXu**
I hereby agree to the terms of the [GreptimeDB CLA](https://gist.github.com/xtang/6378857777706e568c1949c7578592cc)

## What's changed and what's your intention?

Implement create external table parser

## Checklist

- [ ]  I have written the necessary rustdoc comments.
- [x]  I have added the necessary unit tests and integration tests.

## Refer to a related PR or issue link (optional)

#1041

### CODE REVIEW

File: src/frontend/src/instance.rs [hunk 1]
```diff
@@ -663,7 +664,8 @@ pub fn check_permission(
         // database ops won't be checked
         Statement::CreateDatabase(_) | Statement::ShowDatabases(_) | Statement::Use(_) => {}
         // show create table and alter are not supported yet
-        Statement::ShowCreateTable(_) | Statement::Alter(_) => {}
+        Statement::ShowCreateTable(_) | Statement::CreateExternalTable(_) | Statement::Alter(_) => {
+        }
```
Comment (evenyag) [inline src/frontend/src/instance.rs:668]: Does rust fmt allows this?
```suggestion
        Statement::ShowCreateTable(_) | Statement::CreateExternalTable(_) | Statement::Alter(_) => {}
```

**Review by evenyag — APPROVED** [review]
LGTM. Just leave a question about the format

File: src/frontend/src/instance.rs (hunk 1 above)
Comment (WenyXu) [reply src/frontend/src/instance.rs:668]: Yep, it's weird

```bash
~/Projects/greptimedb/ [feat/external-table-parser] cargo fmt --all -- --check 
Diff in /Users/wenyxu/Projects/greptimedb/src/frontend/src/instance.rs at line 664:
         // database ops won't be checked
         Statement::CreateDatabase(_) | Statement::ShowDatabases(_) | Statement::Use(_) => {}
         // show create table and alter are not supported yet
-        Statement::ShowCreateTable(_) | Statement::CreateExternalTable(_) | Statement::Alter(_) => {}
+        Statement::ShowCreateTable(_) | Statement::CreateExternalTable(_) | Statement::Alter(_) => {
+        }
 
         Statement::Insert(insert) => {
             validate_param(insert.table_name(), query_ctx)?;
```

**Comment by WenyXu** [comment]
As discussed in #1272, We still need to add the engine type field.

**Review by MichaelScofield — APPROVED** [review]

**Pull request merged**